### DISCUSSION TITLE
better revisions handling for translations

### DISCUSSION THREAD

**Discussion by wc-matteo** [directus/directus#8159] [Feature Requests]
the [new translation interface](https://github.com/directus/directus/pull/7727) is **AMAZING**. Really really useful to us!

It would be great to have the revisions working with the translations seamlessly, though. Right now it seems the only way to revert changes is to access either the "main" collection or the translations collection or both, according to where the changes to rollback are.

**Comment by rijkvanzanten** [comment]
Totally agree!

We do store the the "parent" of the revision on the revision itself, so we might be able to extract what nested changes happened in the current "main" collection, and offer to revert those as well on the fly 🤔

**Reply by wc-matteo** [reply]
so it could not only handle translations, but all nested relationships in general?

**Reply by rijkvanzanten** [reply]
Technically, yes. In practice, it becomes complicated really fast, as everything can theoretically be connected. We wouldn't want to revert the whole database if you undo a word change in a single article, so we'll have to come up with some way to configure exactly what "paths" to revert, and what to leave as is.

(For example, you changed an article, its translations, and updated the authors description to fix a typo. You'd want to revert the article and its translations, but leave the typo fix on the user)

**Reply by wc-matteo** [reply]
yeah, I think initially the best would be to have it just for the translations. It's the scenario where it makes most sense ux-wise (I think) and it's quite clear cut.

If the rollback was to be applied to all relations, I would let the user choose (in the rollback ui), for every change in a relation, if they wanted to include it in the rollback or not.

**Reply by rijkvanzanten** [reply]
> yeah, I think initially the best would be to have it just for the translations. It's the scenario where it makes most sense ux-wise (I think) and it's quite clear cut.

Sure, but seeing translations is just a regular o2m, there's no reason why it should be arbitrarily limited to fields that just happen to be using a translations UI 🤔

**Reply by wc-matteo** [reply]
talking about ux or implementation?

**Reply by rijkvanzanten** [reply]
Both! I think we could imagine a flow where on revert, we traverse the nested changes, and then present the user with a sort of tree UI saying "These are all the changes applied in this change, which ones would you like to revert"? (maybe with all enabled by default)

**Comment by adiazc9** [comment]
Providing an efficient option to reverse content with translations is essential for multilingual websites.

I think that one of the best options is to show at the bottom of the "Item Revision" section of the parent collection, a list of languages and from there show another "Item Revision/Revision Preview" with the changes for the table languages .

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
